File: programming/java/exemploAula/ExemploEncapsulamento/src/cachorro/Cachorro.java
package cachorro;

public class Cachorro {
    // Atributos
    private String raca;
    private int idade;
    private String nome;
    private String corOlhos;

    public Cachorro(String raca, int idade, String nome, String corOlhos) {
        this.raca = raca;
        this.idade = idade;
        this.nome = nome;
        this.corOlhos = corOlhos;
    }

    // Getters e Setters
    public String getRaca(String usuario, String senha) {
        if (!usuario.equals("admin") || !senha.equals("admin123")) {
            return "Acesso negado. Usuário ou senha incorretos.";
        }
        // Validação de usuario
        return raca;
    }

    public String getCorOlhos() {
        return corOlhos;
    }
    
    public String getNome() {
        return nome;
    }

    public int getIdade() {
        return idade;
    }

    public void setIdade(int idade, String usuario, String senha) {
        if (!usuario.equals("admin") || !senha.equals("admin123")) {
            System.out.println("Acesso negado. Usuário ou senha incorretos.");
            return;
        }

        this.idade = idade;
    }

    public void setNome(String nome) {
        this.nome = nome;
    }

    // Comportamentos (métodos)
    public void ladrar() {
        System.out.println("Au Au");
    }

    public void correr() {
        System.out.println("Correndo...");
    }

    public void dormir() {
        System.out.println("Zzzzzzzzz");
    }

    public void comer() {
        System.out.println("Comendo...");
    }

    public void acordar() {
        System.out.println("Acordando...");
    }
}

File: 02-programacao/php/02-poo/04-mundocao/feitos em aula/polimorfismo/Cachorro.php
<?php

    class Cachorro {

        private string $nome;
        private int $idade;
        public string $raca;
        public string $corOlhos;
        public string $corPelo;

        public function getNome() {
            return $this->nome;
        }

        public function setNome(string $nome) {
            $this->nome = $nome;
        }

        public function getIdade() {
            return $this->idade;
        }

        public function setIdade(int $idade) {
            $this->idade = $idade;
        }

        // Comportamentos / Ações
        public function latir(string $latido){
            echo $latido . PHP_EOL;
        }

        public function comer() {
            echo "Estou comendo..!" . PHP_EOL;
        }

        public function dormir(){
            echo "Estou dormindo com olhos fechados.." . PHP_EOL;
        }

        public function correr(){
            echo "Estou correndo.." . PHP_EOL;
        }

        public function acordar(){
            echo "Estou acordando.." . PHP_EOL;
        }

    }

?>